src/node.cc

```diff
@@ -2475,16 +2477,21 @@ void DLOpen(const FunctionCallbackInfo<Value>& args) {
     mp->nm_context_register_func = NULL;
     mp->nm_modname = mod->modname;
   }
+#endif /* def ENABLE_NAPI */
 
   if (mp == nullptr) {
     uv_dlclose(&lib);
     env->ThrowError("Module did not self-register.");
     return;
   }
 
+#ifdef ENABLE_NAPI
   bool isNapiModule = mp->nm_version == -1;
 
   if (mp->nm_version != NODE_MODULE_VERSION && !isNapiModule) {
+#else /* !defined ENABLE_NAPI */
```

Comment:
This is a case were if we get agreement to enable NAPI by default at compile time, we may need a runtime flag to make sure we don't try to load modules were the version is -1 unless the runtime Flag has enabled NAPI